File: AppBFS.java
public class AppBFS {
    public static void main(String[] args) {
        Graph g = new Graph("tinyG.txt");

        BreadthFirstSearch bfs = new BreadthFirstSearch(g, "0");

        for (String v : g.getVerts()) {
            System.out.print(v + ": (" + bfs.distTo(v) + ") ");
            for (String w : bfs.pathTo(v)) {
                System.out.print(w + " ");
            }
            System.out.println();
        }
    }

}

File: appbfs.cpp
#include <iostream>
#include "graph.h"
#include "breadthfirstsearch.h"

using namespace std;

int main()
{
  Graph g("tinyG.txt");
  BreadthFirstSearch bfs(g, "0");

  for (auto const &v : g.getVerts())
  {
    cout << v << ": ";
    if (bfs.hasPathTo(v))
    {
      for (auto const &w : bfs.pathTo(v))
        cout << w << " ";
    }
    cout << endl;
  }
}
